lib/product-import.coffee

```diff
@@ -365,7 +366,9 @@ class ProductImport
           return @_runReassignmentBeforeCreateOrUpdate(prodToProcess, reassignmentRetries)
 
       # if reassignment is off or if there are no actions which triggers reassignment, run normal update
-      @_updateInBatches(synced.getUpdateId(), updateRequest)
+      return Promise.resolve(@beforeUpdateCallback(existingProducts, prodToProcess, updateRequest))
```

Comment:
Hi @lojzatran. Sorry for late reply. As discussed there are some points that I would address:
- We should not encourage users to mutate passed parameters but return new list of actions instead. In detail it means that the user can either return unmodified list of actions, updated list or an empty list. In case it is an empty list or undefined we should not issue an update (do not trigger update counter too)
- It would be more user friendly if we would pass a list of actions instead of `updateRequest` object which has also other non relevant properties.
- In case user returns an empty list of actions we should not do reassignment. Currently reassignment would be triggered even if we would not do any update.